Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/ScheduledRides/ScheduledRideAssignedOnUpdate.hs

```diff
@@ -104,11 +104,42 @@ sendScheduledRideAssignedOnUpdate Job {id, jobInfo} = withLogTag ("JobId-" <> id
           mbVehicle <- runInReplica $ QVeh.findById driverId
           case (mbDriver, mbVehicle, mbBooking) of
             (Just driver, Just vehicle, Just booking) -> do
-              void $ QDI.updateOnRideAndLatestScheduledBookingAndPickup True Nothing Nothing driverId
-              void $ QRide.updateStatus ride.id DRide.NEW
-              void $ LF.rideDetails ride.id DRide.NEW booking.providerId ride.driverId booking.fromLocation.lat booking.fromLocation.lon
-              void $ sendRideAssignedUpdateToBAP booking ride driver vehicle -- TODO: handle error
-              return Complete
+              let fromLocation = booking.fromLocation
+                  pickupLoc = LatLong {lat = fromLocation.lat, lon = fromLocation.lon}
+                  merchantId = booking.providerId
+                  merchantOperatingCityId = booking.merchantOperatingCityId
+              currentDriverLocation' <- LTF.driversLocation [driverId]
+              mbtransporterConfig <- SCTC.findByMerchantOpCityId ride.merchantOperatingCityId Nothing
+              let isAllJust = (maybe False (\t -> isJust (t.avgSpeedOfVehicle)) mbtransporterConfig) && isJust driverInfo.latestScheduledBooking
```

Comment:
we can use `mbtransporterConfig >>= (.avgSpeedOfVehicle) `
and` let isAllJust = any isNothing [driverInfo.latestScheduledBooking, mbtransporterConfig >>= (.avgSpeedOfVehicle) ]`